packages/server/lib/cookie-jar.ts

```diff
@@ -0,0 +1,51 @@
+import { Cookie, CookieJar as ToughCookieJar } from 'tough-cookie'
+
+interface CookieData {
+  name: string
+  domain: string
+  path?: string
+}
+
+export class CookieJar {
+  _cookieJar: ToughCookieJar
+
+  constructor () {
+    this._cookieJar = new ToughCookieJar(undefined, { allowSpecialUseDomain: true })
+  }
+
+  getAllCookies () {
+    let cookies: Cookie[] = []
+
+    // have to use the internal memstore. looks like an async api, but
+    // it's actually synchronous
+    // @ts-ignore
+    cookieJar.store.getAllCookies((_err, _cookies: Cookie[]) => {
+      cookies = _cookies
+    })
+
+    return cookies
+  }
+
+  setCookie (cookie: string, url: string, sameSiteContext: 'strict' | 'lax' | 'none') {
+    // @ts-ignore
+    this._cookieJar.setCookieSync(cookie, url, { sameSiteContext })
+  }
+
+  removeCookie (cookieData: CookieData) {
+    // have to use the internal memstore. looks like an async api, but
+    // it's actually synchronous
+    // @ts-ignore
+    cookieJar.store.removeCookie(
+      cookieData.domain,
+      cookieData.path || '/',
+      cookieData.name,
+      () => {},
+    )
+  }
+
+  removeAllCookies () {
+    this._cookieJar.removeAllCookiesSync()
```

Comment:
Does this need to be sync?

Reply:
It doesn't need to, but there's no reason for it not to be sync. The cookie store is kept in memory, so all [this ends up doing](https://github.com/salesforce/tough-cookie/blob/b8d751188da697157e5eed81fa1e5f806fdfb541/lib/memstore.js#L148-L151) is basically `cookieJar.cookies = {}`. There's no sync file system access to worry about or anything.